### DISCUSSION TITLE
Welcome to Winter 2024 Computational Mechanics

### DISCUSSION THREAD

**Discussion by cooperrc** [cooperrc/computational-mechanics#216] [Q&A]
Greetings, 
Welcome to our Winter 2024 Computational Mechanics 3-week course! 

We are starting a fast-paced journey to learn to use Python to solve engineering problems. Let's get to know each other this week. 
- _Who_ are you? and _Why_ are you here in ME3255?
- What was your best trick for relaxing last week?
- Show us an example code in Python and tell us what it does. 
    _use the ` ```python - ``` ` markdown format to show code e.g._

````
```python 
x = 1
print(x)
```
````

becomes

```python
x = 1
print(x)
```

**Comment by XanderRaymond** [comment]
Who are you? and Why are you here in ME3255?
Hello, I am Xander and I am a Junior at Uconn. I am in ME3255 to gain a better understanding of how we can use python to solve Mechanical Engineering problems. 
What was your best trick for relaxing last week?
This past week I have been playing Super Smash Bros a lot to relax a bit after the semester. Also a bit of TV and staying up until 4 am. 

```python 
a = int(input("Enter a number for one side: ")) 
b = int(input("Enter a number for the other side: ")) 
c = (a**2 + b**2)**.5
print(c)
```

This code allows you to put in two sides of a triangle to find the hypotenuse for c.

**Reply by cooperrc** [reply]
Smash 64 _old school_ or Ultimate?

Ultimate has been pretty amazing, but I do still have a place in my heart for the original and the mods, [Smash Remix](https://www.ssbwiki.com/Smash_Remix)

**Reply by kunstgarrison** [reply]
I know this is not my post but I have been playing the original recently and have been loving the chaos with  friends

**Reply by XanderRaymond** [reply]
I mostly play ultimate but I have played a bit of Melee. I tried emulating 64 on my computer for a while and got it running but my laptop was struggling for some reason.

**Reply by mim20017** [reply]
I've been playing a lot of brawl with my friends recently!

**Reply by XanderRaymond** [reply]
Brawl was my first smash game and what got me into smash. I used to download a lot of mods for it when I was a bit younger.

**Reply by PAlder23** [reply]
I used to play a lot of Melee when I was younger. A friend recently showed me a program called Slippi where you can play online with your friends or random people. I've been playing a lot recently (Fox main) for nostalgia.

**Reply by XanderRaymond** [reply]
I was using Slippi for a bit too. I suck at melee though so I just played Jigglypuff and fished for Rest.

**Comment by ilh21001** [comment]
Who are you? and Why are you here in ME3255?
_Hi, my name is Ian and i am currently a junior. I am in ME 3255 in order to further my skills in Python, and maybe improve in MatLab as well._

What was your best trick for relaxing last week?
_Taking plenty of breaks and getting up to walk around. Keeping my blood pumping helped me stay sane._ 


Show us an example code in Python and tell us what it does.

_Just a simple albeit famous example. I’m not the best at python but I have used it in ME 3263 and in my thermodynamics class (via coolpropgit)_

```python 
x = ‘hello world’
print(x)
```

**Reply by cooperrc** [reply]
can't argue with the classics

**Comment by hubert-sliwka** [comment]
Who are you? and Why are you here in ME3255?
My name is Hubert and I am currently a senior at UCONN. I am here to get ahead in course work.

What was your best trick for relaxing last week?
Exercising at my gym kept me relaxed and in good spirits last week.

Show us an example code in Python and tell us what it does.
```python 
x = 3
y = 2
z = x + y
print(z)
```
This code simply assigns two numbers as variables and then creates a third variable that adds them and finally prints out the summation.

**Comment by amm20052** [comment]
1. Who are you? and Why are you here in ME3255?
Hello, my name is Audrey. I am currently a junior. I am here in ME 3255 to develop my Python skills. 
2. What was your best trick for relaxing last week?
This past week I have been doing yoga and going on walks outside to help me relax.
3. Show us an example code in Python and tell us what it does.

```
integer = int(input)
if integer % 2 == 0:
        print(f'{integer} is an even number.')
else: 
        print(f'{integer} is an odd number.')
```
This code takes an integer as input, uses the modulus operator to check whether it is even or odd, and then prints the result.

**Reply by cooperrc** [reply]
Yoga is an awesome way to revitalize body+mind, great find. 

In your code, the `int(input)` will force the variable `input` to be an integer. This is equivalent to using `floor` because it removes all the decimal places after the whole number, so `int(2.999) == 2`

**Comment by SuailyTirado** [comment]
Who are you? and Why are you here in ME3255?
_Hello! My name is Suaily and I am a junior at UConn. I am here in hopes of gaining a better understanding of ME and Python working together!_

What was your best trick for relaxing last week?
_Playing a lot of Lethal Company and CS2. Watching too much Netflix :)_

Show us an example code in Python and tell us what it does.
```python
def max(x, y):
   if x > y:
      return x
   else:
      return y

x = 8
y = 3

print(max(x, y))
```
_This code returns the greater number between two integers x and y_

**Reply by cooperrc** [reply]
This will work _almost_ and Python object. Even `strings` have `<` and `>` results.

**Comment by kingjordan3** [comment]
Who are you? and Why are you here in ME3255?
Hi everyone, my name is Jordan and I'm a 3rd year sophomore. I am in ME3255 to learn some Python skills that I can carry into the workforce. Additionally, I am here to get ahead of my coursework. 

What was your best trick for relaxing last week?
This past week I have been relaxing by playing GranTurismo on my racing simulator and by watching and playing basketball. 

Show us an example code in Python and tell us what it does.

```python 

def find_longer_string(x, y):
    if len(x) > len(y):
        return x
    elif len(y) > len(x):
        return y
    else: 
    
x = "Hello"
y = "Python is awesome!"

longer_string = find_longer_string(x, y)
print("The longer string is:", longer_string)
```

This is a function that returns the longer of two strings and then prints the longer string.

**Reply by cooperrc** [reply]
Cool, in `find_longer_string()` you'll end up with no result when `len(x) == len(y)`. What should it return if x and y are `'Python'` and `'python'`?

**Comment by owenlotito** [comment]
Who are you? and Why are you here in ME3255?
My name is Owen Lotito and I am a junior. I am here in ME3255 because it is a requirement for my degree, but I also wanted to get ahead on my courses and keep my mind active throughout winter break. It will be good to get a refresher on Python to get a more solid understanding of it.

What was your best trick to relax over the last week?
My best trick to relax over the last week was to spend time with family and friends who I don't see often when I am at school. I have also been going to the gym and playing video games in my free time to relax. 

An example of python:

```python 
x = 5
y = 8
z = x*y
print(z)```

This simple code assigns two numbers to their respective variable and prints out the product of the two.

**Comment by ksk20011** [comment]
Who are you? Why are you here?

Hello my name is Krish I am a junior, and I am here to refine and develop my somewhat existing python skills in order to implement them into my future educational and professional career.

What did you do to relax?

I just wanted to rest up so I slept a lot, and just watched some movies and caught up on some shows I had fallen behind on

Example of python:

```python 
def filter_even_numbers(numbers):
    even_numbers = [num for num in numbers if num % 2 == 0]
    return even_numbers

number_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


even_numbers_result = filter_even_numbers(number_list)


print(f"The even numbers in the list are: {even_numbers_result}")
```

This code simply filters out the odds numbers from a given list of numbers and prints the remaining even numbers

**Comment by tryonh** [comment]
Who are you? and Why are you here in ME3255?

- My name is Hollister and I am a junior right now, Im in ME3255 to learn about computational mechanics for my major

What was your best trick for relaxing last week?

- The best trick I used for relaxing was going to bed and staying in bed for as long as I wanted because I had no work to do.

Show us an example code in Python and tell us what it does.
use the ```python - ``` markdown format to show code e.g.

```python
r = 2
x = (3.14159)(r*r)
print("area of circle is " x)
```

This code will take a radius and give you the area of a circle from that radius

**Comment by mim20017** [comment]
Who are you? and Why are you here in ME3255?
My name is Michael and I am currently a Junior at UCONN. I am in ME3255 becuase I switched my major from civil to mechanical engineering fairly late, so I play catch up with a few classes.

What was your best trick for relaxing last week?
I played plenty of video games, mainly the Legend of Zelda: Tears of the Kingdom and vampire survivors couch co-op with my friends.

Show us an example code in Python and tell us what it does
```python 
import random

def generate_new_color(color_list):
    if len(color_list) < 2:
        return None

    chosen_colors = random.sample(color_list, 2)

    if 'red' in chosen_colors and 'blue' in chosen_colors:
        new_color = 'purple'
    elif 'red' in chosen_colors and 'yellow' in chosen_colors:
        new_color = 'orange'
    elif 'blue' in chosen_colors and 'yellow' in chosen_colors:
        new_color = 'green'
    else:
        print("Invalid combination. Please ensure you choose two different colors.")
        return None

    return new_color

colors = ['red', 'blue', 'yellow']
new_color = generate_new_color(colors)

if new_color:
    print(f"The new color is: {new_color}")
```
This code is given a list of primary colors and is asked to choose 2 at random, to then ouput a secondary color based on the combination.

**Reply by cooperrc** [reply]
I like the color wheel generator. 

Side note: Python (and other programs) will take colors in the form of [Red, Green, Blue] values. So you could create a the function using addition, 

e.g. 

```python 
red = [1, 0, 0]
blue = [0, 0, 1]

def generate_new_color(color_list):
    new_color = color_list[0] + color_list[1]
    return new_color

>> generate_new_color([red, blue])
[1, 0, 1] # purple
```

**Comment by OwenBass13** [comment]
Who are you? and Why are you here in ME3255?
My name Is Owen and I'm a senior Mechanical Engineering student. I am taking this winter course since I wanted to take an Econ class in the spring for my own personal interests and I wanted to space out my classes so I can take in as much information as I can.

What was your best trick for relaxing last week?
My best trick for relaxing was hanging out with family and friends, watching some movies, and exercising.

Show us an example code in Python and tell us what it does.
use the ```python - ``` markdown format to show code e.g.

```python 
x = 10
t =200
z = t//x
print(z)
```
This code just uses integer division to divide two variables with given inputs

**Reply by cooperrc** [reply]
The `//` returns a whole number. Its an integer _if_ the two inputs are integers, but it will return a float if either `x` or `t` is a float. 

It divides and rounds down to the nearest whole number. If you combine `//` and `%` you can get the whole number & remainder

e.g. 7/4 = 1 & 3/4

```python
>>> 7//4 
1
>>> 7%4
3
```

**Comment by riley-chamberlin** [comment]
Who are you? and Why are you here in ME3255?
Hi, my name is Riley and I am a junior ME major. I am in this class to get ahead on my required courses and learn more about using Python.
What was your best trick for relaxing last week?
To relax this week, I have been watching a lot of sports and spending time with my family.
Show us an example code in Python and tell us what it does
```python
def factorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        return n * factorial(n - 1)

number = 5
result = factorial(number)
print(f"The factorial of {number} is {result}")
```
This code takes an integer as input and outputs the factorial of the integer.

**Reply by cooperrc** [reply]
Excellent use of recursive programming. 

The function `factorial` calls itself within the function. Like Inception for Python functions

**Comment by EasternOT** [comment]
Who are you? and Why are you here in ME3255?
Hello my name is Owen and i'm taking this course to learn how to utilize python.

What was your best trick for relaxing last week?
Spending time with family i haven't seen in awhile.

Show us an example code in Python and tell us what it does.
```python
#winter tution cost calculator
def tuition(x): # where x is the unit of credits
    w=75 # online course fee
    y=30 # tech fee
    z=45 # enrollment fee
    cost=(681*x)+w+y+z
    return cost
tuition(3) # with this input it will give you the cost of a 3 credit class 
```

**Reply by cooperrc** [reply]
hopefully `tuition(x)` will continue to come in handy :)

**Comment by TylerBerendsohn** [comment]
Who are you? and Why are you here in ME3255?
Hi, I'm Tyler Berendsohn and I am a Junior at UConn, I am taking this class start learning about computational mechanics. I am also hoping to get a better understanding of python coding and how it can be used for different situations.

What was your best trick for relaxing last week?
Watching shows, eating good food,  seeing friends and family, as well as working out as often as possible.

Show us an example code in Python and tell us what it does.
```python 
import math
x = 8
y= 12
z= 5

magnitude = math.sqrt((x**2) + (y**2) +(z**2))
unitx = x/magnitude
unity = y/magnitude
unitz = z/magnitude

print("The unit vector is:", (unitx,unity,unitz))
 ```
Given the vector (x, y, z) using any x, y, and z, this code results in a unit vector in the same direction : (unitx, unity, unitz)

**Reply by cooperrc** [reply]
Check out the NumPy method, 
```python
import numpy as np
v = np.array([8, 12, 5])
mag = np.sqrt(np.sum(v**2))
unit = v/mag
print('unit vector is:', unit)
```

**Comment by mattlewis10** [comment]
Who are you? and Why are you here in ME3255?
Hey I'm Matt and I'm here in ME3255 because I need credits.

What was your best trick for relaxing last week?
Laying on my couch doing nothing.

My code:
```python 
x = 3
z = x**2
print(z)
```
This code should hopefully take the value of x and square it, then print it as z.

**Comment by Ewicko** [comment]
**Who are you? and Why are you here in ME3255?**
My name is Ethan Wicko and I'm a senior engineering major. I'm here in ME3255 to hone my python skills and get a better idea of how to apply python scripts to my side projects to get the most engineering out of the time I put in.

**What was your best trick for relaxing last week?**
My best trick for relaxing was to do activities that are fun yet "slowed down time" (slow burn fun). I find that I feel much better at the end of the day if I, for instance, played cards with family for 30 mins (feels like a relatively long amount of time) vs watched an episode of a TV show/YouTube videos for 30 mins (feels like a relatively short amount of time). This makes me feel relaxed and that I am in control of my time!

**Show us an example code in Python and tell us what it does.**
 ```python 
import math

def calculate_area(radius):
    """Calculate the area of a circle with the given radius."""
    area = math.pi * radius**2
    return area

radius = 5
area = calculate_area(radius)
print(f"The area of a circle with radius {radius} is {area:.2f}")
``` 
This code lets you input a radius of a circle and will give you it's radius using a function I made called "calculate_area". It uses the math module to get a value of pi and it rounds the outputted area value to two decimal points.

**Reply by cooperrc** [reply]
Nice docstring use!
```python
   """Calculate the area of a circle with the given radius."""
```

**Comment by thomaspino** [comment]
Who are you? and Why are you here in ME3255?
My name is Thomas Pino and I am a junior mechanical engineering student. I am in ME3255 because I need it for my major and I am also interested in furthering my coding strength.

What was your best trick for relaxing last week?
I worked out and I hung out with my friends and family.

Show us an example code in Python and tell us what it does
```python 
x = 2
y = 3
z = x - y
print(z)
```
This code simply subtracts one variable from the other.

**Comment by andrewcote10** [comment]
Who are you? and Why are you here in ME3255?
Hi, I'm Andrew Cote and I am a junior mechanical engineering student at UConn, I am taking this class since it is a requirement for my major but also to strengthen my python and computational skills.

What was your best trick for relaxing last week?
My best trick for relaxing last week was watching movies and shows.

Show us an example code in Python and tell us what it does.

```python 
x = 14
z = x % 3
print(z)
```

This code takes a value and returns the remainder of that value divided by three using the modulus operator.

**Reply by cooperrc** [reply]
Love the `%`-modulus operator. 

What are your movie/show recommendations?

**Comment by MohamedHaithamAhmed** [comment]
**Who are you? and Why are you here in ME3255?**
My name is Mohamed Haitham Ahmed, and I am a junior mechanical engineering student who will minor in aerospace engineering very soon. I am in ME3255 because I need it for my major as well as to further strengthen my proficiency in coding, as it will allow me to solve engineering problems with greater ease. 

**What was your best trick for relaxing last week?**
I worked out, spent time with the family, cooked for the family, and went out for runs with my friends and family.

**Show us an example code in Python and tell us what it does**

```python 
x = 1
y= 1
z= 9

solution = (x-y+y)*z 
print(solution)
```

This code simply adds and subtracts the variables within the parenthesis then multiplies the resulting variable by the variable outside of the parenthesis, resulting in the the final solution, then it prints out the solution.

**Comment by kunstgarrison** [comment]
Who are you? and Why are you here in ME3255?

I am Garrison Kunst, and I am a Junior ME Major. I am taking this course so I can finish all of my required courses for Mechanical Engineering prior to the start of my senior year.

What was your best trick for relaxing last week?

I spent a lot of time relaxing with my family, something which I haven't been able to do for a while since I could not spend much time with them during Thanksgiving break.

Show us an example code in Python and tell us what it does.
use the ```python - ``` markdown format to show code e.g.

```python - 
# Addition
num1 = 5
num2 = 3
sum_result = num1 + num2
print(f"Addition: {num1} + {num2} = {sum_result}")

# Subtraction
sub_result = num1 - num2
print(f"Subtraction: {num1} - {num2} = {sub_result}")

# Multiplication
mul_result = num1 * num2
print(f"Multiplication: {num1} * {num2} = {mul_result}")

# Division
div_result = num1 / num2
print(f"Division: {num1} / {num2} = {div_result}")

# Exponentiation
exp_result = num1 ** num2
print(f"Exponentiation: {num1} ** {num2} = {exp_result}")``` 



This is just a script to show simple math operations through python, each being labeled individually.

**Comment by PAlder23** [comment]
**Who are you? and Why are you here in ME3255?**

Hi, I'm Paul Alderucci! I am a junior ME major and I am taking this class to get ahead. 

**What was your best trick for relaxing last week?**

My best trick for relaxing last week was to be social and spend time with my friends, as well as going to the gym frequently.

**Show us an example code in Python and tell us what it does.**


```python 
x = float(input("Enter x value: "))
y = float(input("Enter y value: "))
z = float(input("Enter z value: "))

avg = (x + y + z) / 3
print(f"The average of {x}, {y}, and {z} is {avg:.2f}.")
```
This program allows users to input 3 numbers and calculate the average.

**Comment by abepc01** [comment]
**Who are you? and Why are you here in ME3255?**
My name is Abraham Puente and I am a Junior. I am in ME3255 in order to get ahead in classes and advance my understanding of python.

**What was your best trick for relaxing last week?**
I have just enjoyed being around my family and catching up on family drama. 

**Show us an example code in Python and tell us what it does.**
```python
x = 7
y = 8
z = 10
result = (x + y + z) / 3
print(result)
``` 
This code prints the average of x, y, and z.

**Reply by cooperrc** [reply]
Welcome! 

One change to get the average instead of sum, 

```python 
x = 7
y = 8
z = 10
result = x + y + z
print(result/3)
```

**Comment by noah-black1** [comment]
Who are you? and Why are you here in ME3255?
My name is Noah Black, I'm a Junior ME major looking to learn how to use Python to solve engineering problems. 
What was your best trick for relaxing last week?
Going on long bike rides.
Show us an example code in Python and tell us what it does.
``` python -
import math
x = int(input('Enter x value'))
y = int(input('Enter y value'))
z = math.sqrt(x**2 + y**2)
print (f'Z = {z}')
```
The above code solves for the hypotenuse of a triangle with z being the hypotenuse

**Reply by cooperrc** [reply]
> long bike rides

I feel you. definitely my preferred reset and relax method

**Comment by JPardi1** [comment]
Who are you? and Why are you here in ME3255?
My name is Jake Pardi and I am a Junior studying ME. I am in this class to lighten my class load for next year and because it is a requirement for my major. 

What was your best trick for relaxing last week?
I spent my time last week going to the gym, watching netflix, and hanging out with my friends while we are all on break.

Show us an example code in Python and tell us what it does.

```python
x=5
y=8
z=x/y
print(z)
```

This code uses the given variables and divides them to output the new value z.

**Reply by cooperrc** [reply]
This _used_ to return `0` because dividing integers always had to return an integer. 

If you end up on older Python versions e.g. 2.10 I think you have to force `x = 5.0` so the program knows to return a float. 

Fun history lesson!

**Comment by emurphy211** [comment]
Who are you? and Why are you here in ME3255?
 My name is Elizabeth Murphy, and I am a junior ME major. I am taking this class to get ahead in my coursework as well as understand python better. 

What was your best trick for relaxing last week?
    I've been spending a lot of time with family and friends I do not get to see as often! 

Show us an example code in Python and tell us what it does.
```python    
x = 4 
y = 5 
z = x + y
print(z) 
```

This code defines three variables and uses the addition function and print function to calculate and display a number.

**Reply by cooperrc** [reply]
Nice, What number gets displayed?

**Comment by btaps3** [comment]
Who are you? and Why are you here in ME3255?
My name is Brian Tapia-Bernal and I am a Junior Mechanical engineering major. I am here to get one class out of my way for the fall semester in my senior year and strategically manage my class work load for the coming semesters. I also want to understand coding a little bit better so i can improve my work in ME 3264.  

What was your best trick for relaxing last week?
My best trick for relaxing was playing Fortnite all day and going to the gym with my friends and family.
 
```python    
# conversion factor
conv_fac = 0.621371

# calculate miles
miles = kilometers * conv_fac

# printing the result
print('%0.2f kilometers is equal to %0.2f miles' % (kilometers, miles))

```

This code allows the user to enter a distance in kilometers, converts that distance to miles using a standard conversion factor, and then prints the result.

**Comment by NickHelstern** [comment]
**_Who are you? and Why are you here in ME3255?_**
I'm Nick Helstern and I'm currently a Junior ME major at UConn. I'm taking this class both to lighten the load for next semester and because it's a course requirement for my major. I'm also interested in learning more Python and going more in depth with it.  

**_What was your best trick for relaxing last week?_**
Definitely eating a lot of food and being around friends and family. Especially after a long Fall semester it was really nice to come back home and unwind with loved ones. Also had some down time to finally get back into the gym 

**_Show us an example code in Python and tell us what it does:_**
```python
x = 15
y = 20
z = 30
solution = (x + y)*z
print(solution)
```

Although very rudimentary, the above code establishes 3 variables x, y, and z. It is then plugged into an equation given the output solution, which adds x+y then multiplies by z. The solution is then deduced and printed (solution=1050) with the final line of code.

**Comment by leahbrennan20** [comment]
Who are you? and Why are you here in ME3255?
My Name's Leah and I'm a Junior majoring in mechanical engineering. I'm taking ME3255 because it's a requirement for my major and I also want to get ahead, as well as lighten my workload a little bit for the next semester. Taking this course over the winter will also allow me to put more time and focus into coding and getter better at it.

What was your best trick for relaxing last week?
Mostly just spending quality time with friends and family

Show us an example code in Python and tell us what it does.

```python

import numpy as np

x = 32
y = 68

square_root = np.sqrt(x + y)

print(square_root)

```

This code assigns 32 to x and 68 to y. These variables are added, the square root is taken, and the answer is printed.

Forked Repository link: https://github.com/leahbrennan20/computational-mechanics/blob/master/README.md

**Comment by jadongs** [comment]
Who are you? and Why are you here in ME3255?
Hi Guys! My name is Jadon Gomez-Stafford and I'm currently a Junior in Mechanical Engineering with a concentration at Aerospace at UConn. I'm taking this class to strengthen my skills in python and to lighten the load for next semester. Since this class is a course requirement for my major, I think knocking it out now will be a fun and rewarding experience in the winter. I'm very interested in learning a lot in this course!!! 

What was your best trick for relaxing last week?
I finally got to chill and destress from the school semester. I was hanging out with family and friends to escape from the academic mentality. He was nice spending the Holidays with my favorite people 

Show us an example code in Python and tell us what it does:
```python
x = "17"
y = "38"
z = x +y
w = "Fetty Wap once said "
print(w + z)
```
The code basically prints the string "Fetty Wap once said 1738" concatenating them into one.

**Comment by patelved2253** [comment]
**Who are you? and Why are you here in ME3255?**
My Name is Ved Patel and I'm currently a Junior studying Mechanical Engineering with area focus of Aerospace. I am taking this class because it is a Course requirement for my major and taking it now will help lower my class load for senior year.

**What was your best trick for relaxing last week?**
I didn't really do anything out of the ordinary to relax after this past semester, just spent time with family and friends from my hometown.

**Show us an example code in Python and tell us what it does.**
```python 
Length = 5 
Width = 5
Height = 5
Mass = 10
Volume = Length*Width*Height
Density = Mass/Volume
print(Density)
```
The code above can be used to find the density of a uniform object by inputting important values such as is length, width, height, and mass. It will calculate the volume for you and print out what the density of the object is.

**Comment by carlyfresher** [comment]
(1) Who are you? and Why are you here in ME3255? → Hi! I am Carly Fresher and I am a Mechanical Engineering major in my Junior year. I originally signed up for this course because it is a requirement for my major and because I want to get some credits out of the way so that I may have more free time during the semester. However, upon discovering that the course material is heavily centered on Python I am excited to expand my existing Python knowledge over this Winter break session.

(2) What was your best trick for relaxing last week? → I relaxed last week by finally eating some home cooked meals after a long semester of dining hall food and by finally making some time for my favorite hobbies, painting and napping :)

(3) Show us an example code in Python and tell us what it does → The code I included calculates and prints the standard deviation of some provided numbers in list form. Standard deviation is an important statistical tool that quantifies the "spread" or the variation of individual values from the mean of a set of numbers. The code provided utilizes a for loop but there are other ways to manually do this code and a way to utilize the built-in Python function to calculate standard deviation.

```python 
numbers = [3, 6, 1, 4, 5, 8, 3, 5, 8, 1]
avg = sum(numbers) / len(numbers)
squared_difference = 0
for value in numbers:
    difference = value - avg
    squared_difference += difference** 2
variance = squared_difference / len(numbers)
sd = variance ** 0.5
print("The Standard Deviation of the provided data set is", sd)
```

**Comment by vanremenar** [comment]
Hello. My name is Van. I am currently a junior studying mechanical engineering. I am looking to gain more experience in coding and its different applications in the field of engineering.

Since getting home for break, I've found a lot of relaxation in spending time with my family and eating some real food. I've also been happy snowboarding and spending my time with friends.

The code simulates the response of a Magic 8-Ball toy. It prompts the user to ask a question and then provides a random response from a predefined set of answers.

```python 
import random

def magic_8_ball():
    responses = ["Yes, definitely!",
        "No, not a chance.",
        "Ask again later.",
        "Outlook not so good.",
        "You can count on it!",]

    print("Ask the Magic 8-Ball a question:")
    input()
    print("Thinking...")
    print(f"Magic 8-Ball says: {random.choice(responses)}")

magic_8_ball()
```

**Reply by cooperrc** [reply]
Where were you snowboarding? Its been so rainy in New England, didn't think any mountains would be open. 

I love the 8-ball function. Turns out there is a 20-sided dice inside the 8-ball for 20 responses:

[Magic 8 Ball - wiki](https://en.wikipedia.org/wiki/Magic_8_Ball)

● It is certain
● It is decidedly so
● Without a doubt
● Yes definitely
● You may rely on it

● As I see it, yes
● Most likely
● Outlook good
● Yes
● Signs point to yes

● Reply hazy, try again
● Ask again later
● Better not tell you now
● Cannot predict now
● Concentrate and ask again

● Don't count on it
● My reply is no
● My sources say no
● Outlook not so good
● Very doubtful

**Reply by vanremenar** [reply]
I was able to get out to Gore up by lake George for a couple days when I first got home. The conditions were pretty decent before the rain storms, at least for the east coast. More recently, I was just up at Attitash in NH last week, but the rains washed out most of the trails.

**Comment by BendlakJustin** [comment]
**Who are you? and Why are you here in ME3255?**
My name is Justin Bendlak and I am a senior engineering physics major with a concentration in mechanical engineering. I picked this class as an elective since I haven't practiced my python since freshman year and I found it to be an appropriate way to combine some of the skills I've picked up over the years in coding, mathematics, and engineering.

**What was your best trick for relaxing last week?**
I spent a lot of time with family and my hometown friends.

**Show us an example code in Python and tell us what it does.**
````python 
input = [0, 0, 1, 0, 2, 4, 8, 10, 8, 5, 2, 0, 1, 0, 0]
sum_inverse_number = 0
for number in input:
    if number == 0:
        print('zero ignored')
    elif number == 1:
        sum_inverse_number += number
        print('inverse number = number, number added')
    else:
        inverse_number = 1/number
        sum_inverse_number += inverse_number
        print('inverse number added')
print(sum_inverse_number)
 ````
This code takes the sum of the inverses of each number in the input list, ignores zeros, and lets you know what is going on as the loop iterates.

**Reply by cooperrc** [reply]
so this is $\sum \frac{1}{x_i}$ where $x = [1, 2, 4, 8, 10, 8, 5, 2, 1]$?

Careful when you use built-in variables in Python e.g. [`input`](https://docs.python.org/3/library/functions.html#input)

Next time you go to use `input` in the script or the notebook you'll get strange errors. My quick work-around because I'm not very creative-is to add `my_` to the names, 

`my_input`,  `my_array`, `my_def`, ...

**Comment by jackseitz** [comment]
Who are you? and Why are you here in ME3255?
My name is Jack Seitz, and I'm, currently a senior at Stevens Institute of Technology majoring in Mechanical Engineering and minoring in Quantitative Finance. I am taking this class to improve my foundational computational skills in Python. 

What was your best trick for relaxing last week?
I slept in a lot and spent some time with my family over the holiday. 

Show us an example code in Python and tell us what it does.
```python 
x = 4
y = 4
z = x + y
print(z)
```
This code simply adds two variables together and prints the answer; in this case, my favorite number (8)

**Comment by GabeChips** [comment]
> * _Who_ are you? and _Why_ are you here in ME3255?
> My name is Gabriel and I am a senior in biomedical engineering. I'm taking this class as I wanted some more coding experience.
>
> * What was your best trick for relaxing last week?
>Playing some Baldur's Gate 3 seemed to hit the spot but knitting also helped.
>
> * Show us an example code in Python and tell us what it does.
>   _use the `` ```python - ``` `` markdown format to show code e.g._
> 
> ```
> ```python 
> x = int(input('Enter the width of your box'))
> y = int(input('Enter the depth of your box'))
> z = int(input('Enter the height of your box'))
> t = int(input('Enter the thickness of your box'))
> v = ((x * y)* z) - (x - t)*(y - t)*(z - t)
> print(V)
> ```
> ```
> 
> This program will calculate the volume of a box

**Comment by nismod2003** [comment]
Hi I am Nisarg.
Im here to study computational mechanics
Watching netflix
```python 
x = 1
y = 2
T = y-x
print(T)
```

This code will print out the value of T

**Reply by nismod2003** [reply]
https://github.com/cooperrc/computational-mechanics/discussions/216#discussioncomment-8114064
